sdk/go/redis/redis.go

```diff
@@ -23,55 +24,112 @@ func Handle(fn func(payload []byte) error) {
+func (c *Client) Incr(key string) (int64, error) {
+	return incr(c.addr, key)
+}
+
+// Del removes the specified keys. A key is ignored if it does not exist.
+func (c *Client) Del(keys ...string) (int64, error) {
+	return del(c.addr, keys)
 }
 
-// Removes the specified keys. A key is ignored if it does not exist.
-func Del(addr string, keys []string) (int64, error) {
-	return del(addr, keys)
+// Sadd adds the specified values to the set for the specified key, creating
+// it if it does not already exist.
+func (c *Client) Sadd(key string, values ...string) (int64, error) {
+	return sadd(c.addr, key, values)
 }
 
-// Adds the specified values to the set for the specified key, creating it if it
-// does not already exist.
-func Sadd(addr string, key string, values []string) (int64, error) {
-	return sadd(addr, key, values)
+// Smembers gets the elements of the set for the specified key.
+func (c *Client) Smembers(key string) ([]string, error) {
+	return smembers(c.addr, key)
 }
 
-// Get the elements of the set for the specified key.
-func Smembers(addr string, key string) ([]string, error) {
-	return smembers(addr, key)
+// Srem removes the specified elements from the set for the specified key.
+// This has no effect if the key does not exist.
+func (c *Client) Srem(key string, values ...string) (int64, error) {
+	return srem(c.addr, key, values)
 }
 
-// Removes the specified elements from the set for the specified key. This has
-// no effect if the key does not exist.
-func Srem(addr string, key string, values []string) (int64, error) {
-	return srem(addr, key, values)
+// ResultKind represents a result type returned from executing a Redis command.
+type ResultKind uint8
+
+const (
+	ResultKindNil ResultKind = iota
+	ResultKindStatus
+	ResultKindInt64
+	ResultKindBinary
+)
+
+// String implements fmt.Stringer.
+func (r ResultKind) String() string {
+	switch r {
+	case ResultKindNil:
+		return "nil"
+	case ResultKindStatus:
+		return "status"
+	case ResultKindInt64:
+		return "int64"
+	case ResultKindBinary:
+		return "binary"
+	default:
+		return "unknown"
+	}
+}
+
+// GoString implements fmt.GoStringer.
+func (r ResultKind) GoString() string { return r.String() }
+
+// Result represents a value returned from a Redis command.
+type Result struct {
+	Kind ResultKind
+	Val  any
 }
 
-// Run the specified Redis command with the specified arguments, returning zero
-// or more results.  This is a general-purpose function which should work with
-// any Redis command.
-func Execute(addr string, command string, arguments []RedisParameter) ([]RedisResult, error) {
-	return execute(addr, command, arguments)
+// Execute runs the specified Redis command with the specified arguments,
+// returning zero or more results.  This is a general-purpose function which
+// should work with any Redis command.
+//
+// Arguments must be string, []byte, int, int64, or int32.
+func (c *Client) Execute(command string, arguments ...any) ([]*Result, error) {
```

Comment:
I would love some opinions on using an `any` rather than a typed argument. I don't know how I feel about it.

Reply:
Its pretty standard for better or worse :shrug:

Reply:
I would probably have this return an error if an unsupported type is passed in though.

Reply:
Added an error